### PR TITLE
Renaming UpEdges/DownEdges to Predecessors/Successors

### LINKED ISSUE #137: Proper naming for UpEdges/DownEdges methods
Currently, we have an [UpEdges](https://github.com/oras-project/oras-go/blob/main/content/graph.go#L28-L35) method which returns the nodes directly pointing to the current node, and a [DownEdges](https://github.com/oras-project/oras-go/blob/main/content/graph.go#L37-L85) method which returns the nodes directly pointed by the current node.
However, the `UpEdges` / `DownEdges` names may not be appropriate, and we are considering using other names.

The current candidates are:
- UpEdges / DownEdges: 
    - Pros: Currently being used.
    - Cons: There is no academic terms for "up edges" and "down edges" of a Directed Acyclic Graph (DAG).
- Referrers / References: 
    - Pros: Closer to the domain.
    - Cons: The potential concern is that we have too many things named references.
- Parents / Children: 
    - Pros: Straightforward.
    - Cons: DAGs are not trees. 

Let's brainstorm for more names and do a vote!

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Resolves #137.

Signed-off-by: Lixia (Sylvia) Lei <lixia_lei@outlook.com>

### CODE REVIEW

File: target.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ type Target interface {
 	content.TagResolver
 }
 
-// GraphTarget is a CAS with generic tags that supports parent node finding.
+// GraphTarget is a CAS with generic tags that supports predecessor node finding.
```
Comment (shizhMSFT) [inline target.go:26]:
```suggestion
// GraphTarget is a CAS with generic tags that supports direct predecessor node finding.
```

File: content/graph.go [hunk 2]
```diff
@@ -25,32 +25,32 @@ import (
...
-	// UpEdges returns the nodes directly pointing to the current node.
-	UpEdges(ctx context.Context, node ocispec.Descriptor) ([]ocispec.Descriptor, error)
+// PredecessorFinder is an extension of Storage.
+type PredecessorFinder interface {
+	// Predecessors returns the nodes directly pointing to the current node.
+	Predecessors(ctx context.Context, node ocispec.Descriptor) ([]ocispec.Descriptor, error)
 }
 
-// GraphStorage represents a CAS that supports parent node finding.
+// GraphStorage represents a CAS that supports predecessor node finding.
```
Comment (shizhMSFT) [inline content/graph.go:37]:
```suggestion
// GraphStorage represents a CAS that supports direct predecessor node finding.
```

File: registry/remote/repository.go [hunk 3]
```diff
@@ -336,7 +338,7 @@ func (r *Repository) UpEdges(ctx context.Context, desc ocispec.Descriptor) ([]oc
 	return res, nil
 }
 
-// Referrers returns the manifest descriptors directly referencing the given
+// Referrers lists the manifest descriptors directly referencing the given
 // manifest descriptor.
```
Comment (shizhMSFT) [inline registry/remote/repository.go:342]:
```suggestion
// Referrers lists the descriptors of manifests directly referencing the given
// manifest descriptor.
```

File: registry/remote/repository.go [hunk 4]
```diff
@@ -320,10 +320,12 @@ func (r *Repository) tags(ctx context.Context, fn func(tags []string) error, url
 	return parseLink(resp)
 }
 
-// UpEdges returns the manifest descriptors directly referencing the given
+// Predecessors returns the manifest descriptors directly referencing the given
 // manifest descriptor.
```
Comment (shizhMSFT) [inline registry/remote/repository.go:324]:
```suggestion
// Predecessors returns the descriptors of manifests directly referencing the given
// manifest descriptor.
```

File: registry/remote/repository.go [hunk 5]
```diff
@@ -320,10 +320,12 @@ func (r *Repository) tags(ctx context.Context, fn func(tags []string) error, url
...
+// Predecessors internally leverages Referrers, and converts the result artifact
+// descriptors to OCI descriptors.
```
Comment (shizhMSFT) [inline registry/remote/repository.go:326]: You may want to state that this function only works on manifests.

Comment (Wwwsylvia) [reply registry/remote/repository.go]: line #324 mentions this.

**Pull request merged**